rust/main/agents/relayer/src/relayer.rs

```diff
@@ -1408,4 +1508,182 @@ mod test {
+    }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_from_settings_and_run_bad_rpc() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let db_path = temp_dir.path();
+
+        let chains = vec![(
+            KnownHyperlaneDomain::Arbitrum.to_string(),
+            generate_test_chain_conf(
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                None,
+                "http://localhost:9999/rpc",
+            ),
+        )];
+        let origin_chains = &[HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum)];
+        let destination_chains = &[HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum)];
+        let settings =
+            generate_test_relayer_settings(db_path, chains, origin_chains, destination_chains);
+
+        let agent = build_relayer(settings)
+            .await
+            .expect("Failed to build relayer");
+
+        assert!(test_run_relayer(agent).await.is_err());
+    }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_from_settings_and_run_less_destinations() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let db_path = temp_dir.path();
+
+        let chains = vec![
+            (
+                KnownHyperlaneDomain::Arbitrum.to_string(),
+                generate_test_chain_conf(
+                    HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                    None,
+                    "https://sepolia-rollup.arbitrum.io/rpc",
+                ),
+            ),
+            (
+                KnownHyperlaneDomain::Ethereum.to_string(),
+                generate_test_chain_conf(
+                    HyperlaneDomain::Known(KnownHyperlaneDomain::Ethereum),
+                    None,
+                    "https://sepolia-rollup.arbitrum.io/rpc",
+                ),
+            ),
+        ];
+        let origin_chains = &[
+            HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+            HyperlaneDomain::Known(KnownHyperlaneDomain::Ethereum),
+            HyperlaneDomain::Known(KnownHyperlaneDomain::Optimism),
+        ];
+        let destination_chains = &[HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum)];
+        let settings =
+            generate_test_relayer_settings(db_path, chains, origin_chains, destination_chains);
+
+        let agent = build_relayer(settings)
+            .await
+            .expect("Failed to build relayer");
+
+        assert!(test_run_relayer(agent).await.is_err());
+    }
+
+    #[tracing_test::traced_test]
+    #[tokio::test]
+    async fn test_from_settings_and_run_bad_signer() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let db_path = temp_dir.path();
+        let chains = vec![(
+            "arbitrum".to_string(),
+            generate_test_chain_conf(
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                Some(SignerConf::HexKey { key: H256::zero() }),
```

Comment:
does this actually cause issues when building the signer?

Reply:
Yeah, we get
```
2025-06-16T21:44:30.060117Z DEBUG test_from_settings_and_run_bad_signer: relayer::relayer: Relayer startup duration measurement elapsed=5.534138ms event="initialized databases"
2025-06-16T21:44:30.060423Z DEBUG test_from_settings_and_run_bad_signer: relayer::relayer: Relayer startup duration measurement elapsed=221.054µs event="initialized application operation verifiers"
2025-06-16T21:44:30.060585Z ERROR test_from_settings_and_run_bad_signer: relayer::relayer: Critical error when building mailbox err=Building mailbox

Caused by:
   0: Invalid ethereum signer key
   1: crypto error
```